File: telemetry.py
from flask import Flask, request
from influxdb import InfluxDBClient
import os

app = Flask(__name__)

influxdb_host = os.environ.get('INFLUXDB_HOST', 'localhost')
influxdb_port = int(os.environ.get('INFLUXDB_PORT', 8086))
influxdb_database = os.environ.get('INFLUXDB_DATABASE', 'beehive_data')

client = InfluxDBClient(host=influxdb_host, port=influxdb_port)
client.switch_database(influxdb_database)

# Define the API endpoint to receive telemetry data
@app.route('/telemetry', methods=['POST'])
def log_telemetry():
    data = request.json
    json_body = [
        {
            "measurement": "beehive_metrics",
            "tags": {
                "hive_id": data['hive_id']
            },
            "fields": {
                "temperature": data['temperature'],
                "humidity": data['humidity'],
                "audio": data['audio'],
                "bee_count": data['bee_count']
            }
        }
    ]
    client.write_points(json_body)
    return "Data logged", 200

# Run the Flask app
if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000, debug=True)

File: telemetry.js
require('dotenv').config();

const express = require('express');
const bodyParser = require('body-parser');
const { InfluxDB } = require('influx');

const app = express();
const port = process.env.PORT || 5000;

const influx = new InfluxDB({
    host: process.env.INFLUXDB_HOST || 'localhost',
    port: process.env.INFLUXDB_PORT || 8086,
    database: process.env.INFLUXDB_DATABASE || 'beehive_data'
});

// Middleware to parse JSON requests
app.use(bodyParser.json());

// Define the API endpoint to receive telemetry data
app.post('/telemetry', async (req, res) => {
    const data = req.body;

    const points = [
        {
            measurement: 'beehive_metrics',
            tags: {
                hive_id: data.hive_id
            },
            fields: {
                temperature: data.temperature,
                humidity: data.humidity,
                audio: data.audio,
                bee_count: data.bee_count
            }
        }
    ];

    try {
        await influx.writePoints(points);
        res.status(200).send('Data logged');
    } catch (error) {
        console.error('Error writing to InfluxDB:', error);
        res.status(500).send('Internal Server Error');
    }
});



app.listen(port, () => {
    console.log(f"The server is running on {port}");
});
